content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,677 @@
+      },
+      async findUniqueOrThrow({ model, query, args }) {
+        return await customErrorFunc(model, query, args)
+      },
+    },
+  },
+})
+```
+
+### <inlinecode>jsonProtocol</inlinecode> out of Preview!
+
+The `jsonProtocol` preview feature is now Generally Available. When upgrading to Prisma 5, make sure to remove `jsonProtocol` from your preview features, if added.
+
+Prisma 4 and lower:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["jsonProtocol"]
+}
+```
+
+Prisma 5:
+
+```prisma
+generator client {
+  provider = "prisma-client-js"
+}
+```
+
+#### Improved error messages
+
+Due to the switch to the new protocol, several error messages have been improved. For example, the following error message in Prisma 4 and below
+
+```terminal
+Argument id for where.id must not be null. Please use undefined instead.
+```
+
+becomes the following in Prisma 5:
+
+```terminal
+Argument `id` must not be null.
+```
+
+### Removal of array shortcuts
+
+This version drops support for a number of "array shortcuts". These shortcuts were a way to add a single element as a value to an array-based operator instead of wrapping that one element in an array. To make our typings more consistent and logical, we now require array values for these operators.
+
+In most cases, the fix will be as simple as wrapping the existing value in an array. The shortcuts removed in Prisma 5 are:
+
+- [`OR` shortcuts](#or-operators)
+- [`in` and `notIn` shortcuts](#in-and-notin-operators)
+- [PostgreSQL JSON `path` field shortcuts](#path-argument-for-filtering-on-json-fields-in-postgresql)
+- [Scalar list shortcuts](#scalar-lists)
+- [MongoDB Composite list shortcuts](#composite-lists)
+
+While `OR`, `in`, and `notIn` operators are affected, `AND` and `NOT` are not affected by this change.
+
+#### <inlinecode>OR</inlinecode> operators
+
+The following code in Prisma 4 and lower:
+
+```js
+prisma.user.findMany({
+  where: {
+    OR: { email: 'foo@example.com' },
+  },
+})
+```
+
+Will need to be changed to the following in Prisma 5:
+
+```js highlight=3;normal
+prisma.user.findMany({
+  where: {
+    OR: [{ email: 'foo@example.com' }],
+  },
+})
+```
+
+`OR` operators will only accept array values.
+
+#### <inlinecode>in</inlinecode> and <inlinecode>notIn</inlinecode> operators
+
+Similar to `OR`, `in` and `notIn` require array values.
+
+Prisma 4 and lower:
+
+```js
+prisma.user.findMany({
+  where: {
+    id: { in: 123 },
+  },
+})
+
+prisma.user.findMany({
+  where: {
+    id: { notIn: 123 },
+  },
+})
+```
+
+Prisma 5:
+
+```js highlight=4,12;normal
+prisma.user.findMany({
+  where: {
+    id: {
+      in: [123],
+    },
+  },
+})
+
+prisma.user.findMany({
+  where: {
+    id: {
+      notIn: [123],
+    },
+  },
+})
+```
+
+<details><summary>Suggestion for single elements</summary>
+
+If your `in` and `notIn` values are only one element, you can also update your code to not use these operators at all:
```

Comment:
Same is true for OR, right?

Reply:
@SevInf could you let me know here?

Reply:
A 1 element `OR` indeed does nothing as far as I know.